CMake/HPHPCompiler.cmake

```diff
@@ -175,6 +175,28 @@ if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" OR ${CMAKE_CXX_COMPILER_ID} STREQU
       list(APPEND GENERAL_CXX_OPTIONS "mcrc32")
     endif()
 
+    # ARM64
+    if(IS_AARCH64)
+      # Force char type to be signed, which is not the case on aarch64.
+      list(APPEND GENERAL_OPTIONS "fsigned-char")
+
+      # If a CPU was specified, build a -mcpu option for the compiler.
+      set(CPU "" CACHE STRING "CPU to tell gcc to optimize for (-mcpu)")
```

Comment:
As a side note, when fixing this up, use an `OPTION` instead of a normal variable. (yes, I know my code for the MSVC options doesn't use them, I should fix that >_>...)

Also, don't name it `CPU`, as it's too ambiguous as-to what the option actually does. For the MSVC section, I named the similar, but not exact same, option `MSVC_FAVORED_ARCHITECTURE`. Perhaps `AARCH64_TARGET_ARCHITECTURE`?